Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -524,12 +675,37 @@ extension Double: BSONValue {
 }
 
 /// An extension of `Int` to represent the BSON Int32 or Int64 type.
-/// The `Int` will be encoded as an Int32 if possible, or an Int64 if necessary.
-extension Int: BSONValue {
-    public var bsonType: BSONType { return self.int32Value != nil ? .int32 : .int64 }
+/// On 64-bit systems, `Int` corresponds to a BSON Int64. On 32-bit systems, it corresponds to a BSON Int32.
+extension Int: BSONNumber {
+    /// `Int` corresponds to a BSON int32 or int64 depending upon whether the compilation system is 32 or 64 bit.
+    internal static var bsonType: BSONType {
+        return MemoryLayout<Int>.size == 4 ? .int32 : .int64
```

Comment:
Should this be reused in the tests to replace [this code](https://github.com/mongodb/mongo-swift-driver/blob/master/Tests/MongoSwiftTests/TestUtils.swift#L49)?

Reply:
I updated the code you linked to use `Int`'s bsonType to check whether or not the system is 32-bit (as is now done elsewhere in this PR), but I kept the `is32Bit` variable since it is a little more explicit about what it's testing and makes our tests that use it a little easier to read.